module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,72 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <cctype>
+#include <iostream>
+#include <iterator>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(
+    const std::array<std::array<uint8_t, width>, height>& bitmap) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressedBitmap{};
```

Comment:
Maybe reserve memory, when we can predict size? You use shirnk_to_fit at the end.